pycheribuild/projects/cross/gdb.py

```diff
@@ -80,8 +80,8 @@ class BuildGDBBase(CrossCompileAutotoolsProject):
     @classmethod
     def dependencies(cls, config: CheriConfig) -> "list[str]":
         deps = super().dependencies(config)
-        # For the native-hybrid build gmp must be installed via ports.
-        if not cls.get_crosscompile_target().is_cheri_hybrid():
+        # For the native build gmp must be installed via ports.
+        if cls.get_crosscompile_target().is_cheri_hybrid():
```

Comment:
I think this fixes your use case but breaks Alex's gdb-native-hybrid build?

Reply:
Hmm, is this to build natively on a Morello host?  I'm not sure what the right condition to check is.  Previously it enabled "gmp" for "gdb-native" but not for targets like "gdb-riscv64-hybrid" or "gdb-risc64-hybrid-for-purecap-rootfs".  I think what we want is to enable "gmp" for all but "gdb-native" and "gdb-native-hybrid".

Reply:
Probably `if not (cls.get_crosscompile_target().is_native() and cls.get_crosscompile_target().is_cheri_hybrid())`

Reply:
But yes; the reason is that there's no NATIVE_HYBRID (which is what NATIVE_NON_PURECAP is on a Morello host) gmp target (but maybe that would be the better approach?..)

Reply:
I think that's what I originally tried but I think I ran into problems with conflicting install paths and just decided to use the ports versions instead on native cheribsd.